src/proxy/src/write.rs

```diff
@@ -506,7 +506,35 @@ impl Proxy {
             .await?;
 
         let mut success = 0;
-        for insert_plan in plan_vec {
+
+        // TODO: concurrently run the insert plan here
+        for plan in plan_vec {
+            // check limit first
+            // TODO: if one table is blocked, maybe should not lead to failure of whole
+            // batch?
+            self.instance
+                .limiter
+                .try_limit(&plan)
+                .box_err()
+                .context(ErrWithCause {
+                    code: StatusCode::INTERNAL_SERVER_ERROR,
+                    msg: "table is blocked",
+                })?;
+
+            let insert_plan = match plan {
```

Comment:
This check seems unnecessary, we could move this `try_limit` inside `write_request_to_insert_plan`, so you don't need to do this.

Reply:
I think the steps should be:
- build plan
- check plan
- execute the valid plan

It maybe  strange to mixed build step and check step?